// @progress/kendo-react-grid/GridComponent.mjs
/**
 * @license
 *-------------------------------------------------------------------------------------------
 * Copyright © 2025 Progress Software Corporation. All rights reserved.
 * Licensed under commercial license. See LICENSE.md in the package root for more information
 *-------------------------------------------------------------------------------------------
 */
import * as a from "react";
import { uGrid as Tt, hasValidLicense as Pt, getLicenseMessage as Kt, validatePackage as At, getter as v, classNames as f, WatermarkOverlay as Ue } from "@progress/kendo-react-common";
import { process as Mt } from "@progress/kendo-data-query";
import { combineFilters as Bt, getDetailExpandableOptions as Lt, getGroupExpandableOptions as Ft, getSelectionOptions as zt, tableKeyboardNavigationTools as z, Pager as Ot, tableKeyboardNavigationBodyAttributes as _t, tableKeyboardNavigationScopeAttributes as qe } from "@progress/kendo-react-data-tools";
import { GridHierarchyCell as Vt } from "./cells/hierarchycell/GridHierarchyCell.mjs";
import { GridEditCell as Wt } from "./cells/editcell/GridEditCell.mjs";
import { Header as $t } from "./header/Header.mjs";
import { HeaderRow as jt } from "./header/HeaderRow.mjs";
import { FilterRow as Ut } from "./header/FilterRow.mjs";
import { GroupPanel as qt } from "./header/GroupPanel.mjs";
import { Footer as Jt } from "./footer/Footer.mjs";
import { isRowReorderEnabled as Qt, getRowSpanOptions as Je, getColumnWidth as re, isSorted as Qe, flatData as Xt, autoGenerateColumns as Yt, mapColumns as Zt, clientColumn as pt, getFlatColumnsState as el, getColumnState as ne, getColSpan as tl, resolveCells as ll, readColumns as al } from "./utils/index.mjs";
import { GridCell as rl } from "./cells/datacell/GridCell.mjs";
import { GridGroupCell as Xe } from "./cells/groupcell/GridGroupCell.mjs";
import { GridRow as nl } from "./rows/GridRow.mjs";
import { GridHeaderSelectionCell as il } from "./header/GridHeaderSelectionCell.mjs";
import { GridNoRecords as Ye } from "./components/noRecords/GridNoRecords.mjs";
import { operators as Ze } from "./filterCommon.mjs";
import { FooterRow as ol } from "./footer/FooterRow.mjs";
import { normalize as dl } from "./paging/GridPagerSettings.mjs";
import { packageMetadata as ie } from "./package-metadata.mjs";
import { GridDetailCell as cl } from "./cells/detailcell/GridDetailCell.mjs";
import { GridDetailHierarchyCell as ml } from "./cells/GridDetailHierarchyCell.mjs";
import { GridNoRecordsContainer as pe } from "./components/noRecords/GridNoRecordsContainer.mjs";
import { GridClientWrapper as et } from "./GridClientWrapper.mjs";
import { GridColGroup as sl } from "./components/colGroup/GridColGroup.mjs";
import { GridTable as ul } from "./components/table/GridTable.mjs";
import { GridDropClue as tt } from "./components/GridDropClue.mjs";
import { GridDragClue as lt } from "./components/GridDragClue.mjs";
import { GridTableBody as at } from "./components/table/GridTableBody.mjs";
import { PagerContainer as fl } from "./components/PagerContainer.mjs";
import { GridTableScrollable as gl } from "./components/table/GridTableScrollable.mjs";
import { GridElementContainer as rt } from "./components/GridElementContainer.mjs";
import { GridContainerElementContainer as bl } from "./components/GridContainerElementContainer.mjs";
import { VirtualScrollHeightContainer as hl } from "./components/VirtualScrollHeightContainer.mjs";
import { gridAriaLabel as nt, messages as yl } from "./messages/index.mjs";
import { GridReorderableRowsContainer as Cl } from "./components/GridDraggableRowsContainer.mjs";
import { GridRowReorderCell as wl } from "./cells/rowreordercell/GridRowReorderCell.mjs";
import { GridHeaderRowReorderCell as El } from "./header/client/GridHeaderRowReorderCell.mjs";
import { GridLoader as it } from "./components/GridLoader.mjs";
import { gridPremiumFeatures as Il } from "./utils/premium.mjs";
import { LocalizationService as Sl, IntlService as vl } from "@progress/kendo-react-intl";
import { getVirtualCellsToRender as Rl } from "./utils/virtualColumns.mjs";
import { GridEditDialog as Gl } from "./components/GridEditDialog.mjs";
import { GridCellServer as xl } from "./cells/datacell/GridCellServer.mjs";
import { GridEditCellServer as kl } from "./cells/editcell/GridEditCellServer.mjs";
import { GridSelectionCellServer as Nl } from "./cells/selectioncell/GridSelectionCellServer.mjs";
import { GridSelectionCell as Dl } from "./cells/selectioncell/GridSelectionCell.mjs";
import { GridHierarchyCellServer as Hl } from "./cells/hierarchycell/GridHierarchyCellServer.mjs";
import { GridRowReorderCellServer as Tl } from "./cells/rowreordercell/GridRowReorderCellServer.mjs";
import { GridDetailCellServer as Pl } from "./cells/detailcell/GridDetailCellServer.mjs";
import { GridGroupCellServer as Kl } from "./cells/groupcell/GridGroupCellServer.mjs";
const Al = a.forwardRef((e, ot) => {
  var Te, Pe, Ke, Ae, Me, Be, Le, Fe, ze, Oe, _e, Ve;
  const T = e.id + "-role-element-id", x = e.navigatable ? T : "";
  let R = e.columnsState || [];
  const dt = (t, n, i, d, o, l, c) => {
    const s = [], h = Xt(
      s,
      t,
      n,
      { index: i },
      d !== void 0,
      o,
      l,
      St.defaultExpand,
      c
    );
    return { flattedData: s, resolvedGroupsCount: h };
  }, ct = (t) => {
    const n = t.filter(
      (i) => i && i.type && i.type.displayName === "KendoReactGridColumn"
    );
    return al(n, R, { prevId: 0, idPrefix: x });
  }, mt = () => {
    const t = [], n = (i, d) => i == null ? void 0 : i.forEach((o) => {
      const l = o.hidden || d;
      t.push({
        ...o,
        hidden: l
      }), n(o.children, l);
    });
    return n(R, !1), t;
  }, st = (t) => {
    const n = mt();
    return t.filter((i) => {
      var d;
      return !i.hidden && !((d = n.find((o) => o.id === i.id)) != null && d.hidden);
    });
  }, ut = (t, n) => {
    m = ct(t), m.filter((i) => !i.hidden).length === 0 && (m = Yt(
      C,
      e.group,
      {
        column: K.column
      },
      {
        prevId: 0,
        idPrefix: x
      }
    )), gt(m, n), m = st(m), P = Zt(m, !0), ce = m.map(pt);
  }, ft = (t, n) => {
    const i = (d) => {
      var l;
      const o = n.find((c) => c.id === d.id);
      return o ? (o.children = (l = d.children) == null ? void 0 : l.map(i), o) : ne(d);
    };
    R = t.filter((d) => d.parentIndex === -1).map(i);
  }, gt = (t, n) => {
    t.filter((l) => l.columnType === "checkbox").forEach((l) => {
      l.width = l.width || "50px", l.defaultCell = e.isClient ? Dl : Nl, l.defaultHeaderCell = il, l._type = "edit";
    }), N !== void 0 && t.filter((l) => l.columnType === "reorder").forEach((l) => {
      l.width = l.width || "50px", l.defaultCell = e.isClient ? wl : Tl, l.defaultHeaderCell = El, l.sortable = !1, l.filterable = !1, l.editable = !1;
    });
    const i = el(R);
    ft(t, i);
    const d = {
      id: "",
      resizable: !0,
      width: "32px",
      title: " ",
      declarationIndex: -1,
      orderIndex: -1,
      children: [],
      parentIndex: -1,
      depth: 0,
      colSpan: 0,
      headerColSpan: 0,
      rowSpan: 0,
      left: 0,
      right: 0,
      index: 0,
      rightBorder: !1,
      ariaColumnIndex: 0,
      isAccessible: !0
    };
    let o = 0;
    if (K.enabled && e.detail) {
      const l = {
        ...d,
        _type: "expand",
        id: z.generateNavigatableId(`${o++}`, "expand", "column"),
        defaultCell: e.isClient ? Vt : Hl,
        field: K.column,
        headerClassName: f(u.hierarchyCell({}))
      };
      t.unshift(l), R.unshift(i.find((c) => c.id === l.id) || ne(l));
    }
    for (let l = 0; l < n; l++) {
      const c = {
        ...d,
        isAccessible: !1,
        defaultCell: e.isClient ? Xe : Kl,
        id: z.generateNavigatableId(`${o++}`, "group", "column"),
        field: "value",
        locked: e.lockGroups
      };
      t.unshift(c), R.unshift(i.find((s) => s.id === c.id) || ne(c));
    }
    t.slice(o).forEach((l) => {
      l.parentIndex >= 0 && (l.parentIndex += o), l.rowSpannable = l.rowSpannable !== void 0 ? Je(l.rowSpannable) : ge;
    });
  }, oe = () => e.isClient ? m : ce, de = (t) => {
    const n = [];
    let i = null, d = 0;
    if (b.forEach((o, l) => {
      const c = parseFloat((o.width || "").toString()) || 10;
      if (d) {
        d--, i && (i.width += c);
        return;
      }
      const s = Math.min(tl(o, t), b.length - l);
      d = s - 1, i = {
        width: c,
        colSpan: s,
        columnIndex: l
      }, n.push(i);
    }), e.columnVirtualization) {
      const o = ht.current || 0, l = yt.current || parseFloat(((e.style || {}).width || "").toString());
      return Rl({
        cellModels: n,
        columns: b,
        tableViewPortWidth: l,
        scrollLeft: o
      });
    }
    return n;
  }, bt = () => {
    const { pageable: t, take: n, pageSize: i } = e;
    if (!E)
      return 0;
    if (!t) {
      if (n)
        return n;
      if (i)
        return i;
    }
    const d = e.rowHeight || wt.current, o = Ct.current;
    return o && d ? Math.ceil(o / d * 1.5) : 0;
  };
  let g = [], j, m = [], ce = [], P = [[]];
  const ht = e.scrollLeftRef || { current: 0 }, yt = e.widthRef || { current: 0 }, Ct = e.containerHeightRef || { current: 0 }, wt = e.minRowHeightRef || { current: 0 }, O = e.localization || new Sl(e.language), Et = e.intl || new vl((Te = e.locale) != null ? Te : "en"), k = e.unstyled, u = k && k.uGrid ? k.uGrid : Tt, N = Qt(e.rowReorderable), E = e.scrollable === "virtual" || e.scrollable === void 0 && e.isClient && Pt(ie) || !1, me = Kt(ie), D = e.autoProcessData === !0 ? {
    group: !0,
    sort: !0,
    filter: !0,
    search: !0,
    page: !0
  } : e.autoProcessData;
  let C, G;
  if (Array.isArray(e.data) ? (C = e.data, G = (Pe = e.total) != null ? Pe : C.length) : (C = ((Ke = e.data) == null ? void 0 : Ke.data) || [], G = (Be = (Me = e.total) != null ? Me : (Ae = e.data) == null ? void 0 : Ae.total) != null ? Be : C.length), D) {
    const t = D.page && !(E && !e.pageable), { data: n, total: i } = Mt(C, {
      group: D.group ? e.group : void 0,
      sort: D.sort ? e.sort : void 0,
      filter: Bt(
        D.filter ? e.filter : void 0,
        D.search ? e.search : void 0
      ),
      ...t ? {
        take: e.pageable ? e.take || 10 : e.take,
        skip: e.skip || 0
      } : {}
    });
    C = n, G = (Le = e.total) != null ? Le : i;
  }
  const _ = C.length === G, { size: I = "medium" } = e, It = typeof e.groupable == "object" && e.groupable.footer || "none", V = e.groupable === !0 || typeof e.groupable == "object" && e.groupable.enabled !== !1, K = Lt(!!e.detail), St = Ft(
    typeof e.groupable == "object" && e.groupable.enabled !== !1 ? e.groupable.expandable : e.groupable
  ), se = !!((Fe = e.group) != null && Fe.length), { resolvedGroupsCount: vt, flattedData: Rt } = dt(
    C,
    It,
    _ ? 0 : e.skip || 0,
    e.group,
    e.detailExpand,
    e.groupExpand,
    e.dataItemKey
  );
  g = Rt;
  const U = bt(), q = ((ze = e.virtualSkipRef) == null ? void 0 : ze.current) || 0;
  let ue = G;
  if (E) {
    let t = e.skip || 0;
    if ((se || e.pageable) && (t = q, ue = g.length), _ || se || e.pageable)
      if (U === 0) {
        const n = Math.min(e.take || e.pageSize || 20, g.length);
        j = g.slice(t, t + n);
      } else
        j = g.slice(t, t + U);
  }
  const fe = zt(e.selectable), ge = Je(e.rowSpannable), Gt = fe && fe.drag ? "none" : void 0, W = a.useMemo(() => a.Children.toArray(e.children), [e.children]);
  ut(W, vt);
  const J = a.useMemo(() => {
    const t = Il(e, m);
    return {
      premium: t.length > 0,
      features: t
    };
  }, [e, m]), be = a.useMemo(() => J.premium ? !At(ie, { component: "Grid", features: J.features }) : !1, [J.premium]), he = W.map((t) => t && t.type && t.type.displayName === "KendoReactGridToolbar" ? a.cloneElement(t, { ...t.props, _ariaControls: T, ariaLabel: "Top toolbar" }) : null), $ = W.filter((t) => t && t.type && t.type.displayName === "KendoReactGridNoRecords"), ye = W.filter(
    (t) => t && t.type && t.type.displayName === "KendoReactGridStatusBar"
  ), b = m.filter((t) => t.children.length === 0), Ce = V && /* @__PURE__ */ a.createElement(qt, { columns: oe(), group: e.group || [], ariaControls: T }), we = /* @__PURE__ */ a.createElement(
    $t,
    {
      size: I,
      staticHeaders: e.scrollable !== "none",
      draggable: e.reorderable || V,
      headerRow: /* @__PURE__ */ a.createElement(
        jt,
        {
          cells: e.cells,
          sort: e.sort,
          sortable: e.sortable,
          group: e.group || [],
          groupable: V,
          filter: e.filter,
          filterable: e.filterable,
          filterOperators: e.filterOperators || Ze,
          columnMenu: e.columnMenu,
          columnMenuIcon: e.columnMenuIcon,
          columns: m,
          columnsMap: P,
          navigatable: !!e.navigatable,
          localization: O,
          unstyled: k,
          headerSelectionValue: !!(e.select && g.filter((t) => t.rowType === "data").every(
            (t) => e.select && e.dataItemKey && v(e.dataItemKey)(t.dataItem) !== void 0 ? e.select[v(e.dataItemKey)(t.dataItem)] : void 0
          ))
        }
      ),
      filterRow: e.filterable && /* @__PURE__ */ a.createElement(
        Ut,
        {
          cells: e.cells,
          size: I,
          columns: m,
          filter: e.filter,
          filterOperators: e.filterOperators || Ze,
          sort: e.sort,
          navigatable: !!e.navigatable,
          ariaRowIndex: P.length + 1,
          localization: O
        }
      ) || void 0,
      cols: b.map((t, n) => /* @__PURE__ */ a.createElement("col", { key: n.toString(), width: re(t) }))
    }
  ), Ee = b.findIndex((t) => typeof t.colSpan == "function") > -1;
  let Q;
  Ee || (Q = de(null));
  const X = ((Oe = e.editable) == null ? void 0 : Oe.mode) === "dialog", xt = (_e = e.editable) == null ? void 0 : _e.enabled, kt = (t, n, i, d, o) => {
    let l = !1;
    const c = e.select && e.dataItemKey && v(e.dataItemKey)(t.dataItem) !== void 0 ? e.select[v(e.dataItemKey)(t.dataItem)] : void 0, s = e.highlight && e.dataItemKey && v(e.dataItemKey)(t.dataItem) !== void 0 ? e.highlight[v(e.dataItemKey)(t.dataItem)] : void 0;
    return Ee && (Q = de(t.dataItem)), {
      row: Q.map(({ columnIndex: h, colSpan: H }) => {
        var F, We, $e, je;
        const r = b[h];
        let w;
        if ((F = r.rowSpannable) != null && F.enabled && t.rowType === "data" && r.field && o) {
          const S = r.field ? ($e = (We = r.rowSpannable).valueGetter) == null ? void 0 : $e.call(We, t.dataItem, r.field) : null;
          w = { value: S, count: 1 }, o[r.field] && ((je = o[r.field]) == null ? void 0 : je.value) === S && o[r.field] !== null ? (o[r.field].count++, w.count = null) : o[r.field] = w;
        }
        const y = r.id ? r.id : h, ee = f(u.contentSticky({ locked: r.locked }), r.className), te = r.left !== void 0 ? { left: r.left, right: r.right } : {};
        let M = !1;
        if (r.editable && e.editable) {
          const S = e.edit && e.dataItemKey ? e.edit[v(e.dataItemKey)(t.dataItem)] : void 0, ae = typeof S == "boolean" ? S : Array.isArray(S) ? r.field && S.indexOf(r.field) > -1 : r.field !== void 0 && S === r.field;
          ae && r.columnType === "data" && (ae === !0 || ae === r.field) && (l = !0, X || (M = !0));
        }
        const B = r._type === "expand", L = {
          locked: e.lockGroups,
          id: z.generateNavigatableId(
            `${n}-${String(h)}`,
            x,
            B || t.rowType === "groupHeader" || t.rowType === "groupFooter" || r.field === "value" ? "nodata" : "cell"
          ),
          colSpan: H,
          dataItem: t.dataItem,
          field: r.field,
          editor: r.editor,
          format: r.format,
          columnType: r.columnType,
          rowReorderable: N,
          className: ee,
          cells: ll(e.cells, r.cells),
          columnIndex: h,
          columnsCount: b.length,
          rowType: t.rowType,
          level: t.level,
          expanded: t.expanded,
          dataIndex: t.dataIndex,
          rowDataIndex: i,
          columnPosition: te,
          style: {},
          ariaColumnIndex: r.ariaColumnIndex,
          isSelected: !!((r == null ? void 0 : r._type) === "edit" ? c : Array.isArray(c) && c.indexOf(h) > -1),
          isHighlighted: !!((r == null ? void 0 : r._type) === "edit" ? s : Array.isArray(s) && s.indexOf(h) > -1),
          isSorted: !!Qe(r.field, e.sort),
          isInEdit: M,
          isAlt: d,
          unstyled: k,
          group: t.group,
          localization: O,
          intl: Et,
          _rowSpan: w
        }, le = r.defaultCell || M && (e.isClient ? Wt : kl) || (e.isClient ? rl : xl);
        return /* @__PURE__ */ a.createElement(le, { key: y, cellProps: L });
      }),
      isInEdit: l,
      isSelected: typeof c == "boolean" && c,
      isHighlighted: typeof s == "boolean" && s
    };
  };
  let Nt = 0, Y = null;
  const Ie = (t) => t >= g.length - Nt, Z = [], Se = !g.length;
  let A = 0;
  if (g.length) {
    const t = P.length + (e.filterable ? 1 : 0) + 1;
    let n = e.skip || 0, i = -1, d = 0;
    const o = ge.enabled ? {} : void 0;
    if (E) {
      if (q > 0) {
        const l = g.slice(0, q).filter((c) => c.rowType === "data").length;
        i += l, n += l;
      }
      !e.pageable && _ && (i += e.skip || 0);
    }
    (j || g).forEach((l, c) => {
      l.rowType === "data" && i++;
      const s = l.dataIndex % 2 !== 0, h = e.dataItemKey && v(e.dataItemKey)(l.dataItem), H = c + n, r = h || "ai" + H, w = r + "_1", y = kt(l, r, i, s, o);
      if (A = H + t + d, X && xt && y.isInEdit && (Y = l.dataItem), Z.push(
        /* @__PURE__ */ a.createElement(
          nl,
          {
            key: r,
            dataItem: l.dataItem,
            isAltRow: s,
            isInEdit: y.isInEdit && !X,
            rowType: l.rowType,
            isRowReorderable: N,
            isHidden: Ie(c),
            onClick: null,
            onDoubleClick: null,
            rowHeight: e.rowHeight,
            ariaRowIndex: A,
            absoluteRowIndex: H,
            dataIndex: i,
            isSelected: y.isSelected,
            isHighlighted: y.isHighlighted,
            rows: e.rows
          },
          y.row
        )
      ), e.detail && l.rowType === "data" && l.expanded) {
        const ee = b.length - (K.enabled ? 1 : 0) - (e.group ? e.group.length : 0) || 1;
        d++, A = H + t + d;
        const te = e.isClient ? cl : Pl;
        Z.push(
          /* @__PURE__ */ a.createElement(
            "tr",
            {
              key: w,
              className: f(u.detailTr({ isAlt: s })),
              style: {
                visibility: Ie(c) ? "hidden" : "",
                height: e.detailRowHeight
              },
              role: "row",
              "aria-rowindex": A
            },
            e.group && e.group.map((M, B) => {
              var F;
              const L = (F = y == null ? void 0 : y.row[B]) == null ? void 0 : F.props.style, le = L ? { left: L.left, right: L.right } : {};
              return /* @__PURE__ */ a.createElement(
                Xe,
                {
                  key: B,
                  cellProps: {
                    id: "",
                    dataItem: l.dataItem,
                    field: M.field,
                    dataIndex: l.dataIndex,
                    columnPosition: le,
                    style: {},
                    ariaColumnIndex: 1 + B,
                    isSelected: !1,
                    locked: e.lockGroups,
                    cells: e.cells,
                    group: l.group
                  }
                }
              );
            }),
            K.enabled && /* @__PURE__ */ a.createElement(
              ml,
              {
                unstyled: k,
                id: z.generateNavigatableId(`${w}-dhcell`, x)
              }
            ),
            /* @__PURE__ */ a.createElement(
              te,
              {
                dataItem: l.dataItem,
                dataIndex: l.dataIndex,
                colSpan: ee,
                ariaColIndex: 2 + (e.group ? e.group.length : 0),
                detail: e.detail,
                id: z.generateNavigatableId(`${w}-dcell`, x)
              }
            )
          )
        );
      }
    });
  }
  const ve = {
    size: I,
    total: G,
    skip: e.skip || 0,
    take: (e.take !== void 0 ? e.take : e.pageSize) || 10,
    ...dl(e.pageable || {})
  }, Re = /* @__PURE__ */ a.createElement(fl, null, e.pager ? /* @__PURE__ */ a.createElement(e.pager, { ...ve }) : /* @__PURE__ */ a.createElement(Ot, { className: f(u.pager({})), ...ve })), Dt = (t, n) => /* @__PURE__ */ a.createElement("col", { key: n.toString(), width: re(t) }), Ge = (Ve = e.cells) != null && Ve.footerCell || m.some((t) => {
    var n;
    return !!((n = t.cells) != null && n.footerCell);
  }) ? /* @__PURE__ */ a.createElement(
    Jt,
    {
      size: I,
      staticHeaders: e.scrollable !== "none",
      row: /* @__PURE__ */ a.createElement(
        ol,
        {
          cells: e.cells,
          idPrefix: x,
          columns: m,
          ariaRowIndex: A + 1
        }
      ),
      cols: b.map(Dt)
    }
  ) : null, xe = /* @__PURE__ */ a.createElement(sl, null, b.map((t, n) => /* @__PURE__ */ a.createElement(
    "col",
    {
      key: n.toString(),
      className: Qe(t.field, e.sort) ? f(u.sorted({})) : void 0,
      width: re(t)
    }
  ))), ke = e.reorderable || V, { detail: Ml, cells: Bl, rows: Ll, ...Ht } = e, Ne = /* @__PURE__ */ a.createElement(
    "tbody",
    {
      role: "rowgroup",
      className: f(u.tbody({})),
      ..._t
    },
    Z
  ), De = Y ? /* @__PURE__ */ a.createElement(Gl, { columns: b, customEditDialog: e.editDialog, dataItem: Y }) : null;
  let p = Ne;
  N && (p = /* @__PURE__ */ a.createElement(
    Cl,
    {
      unstyled: u,
      columns: m,
      rowReorderSettings: e.rowReorderable
    },
    Ne
  ));
  const He = {
    gridRef: ot,
    innerGrid: e.innerGrid,
    gridProps: Ht,
    columnsRef: oe(),
    columnsMapRef: P,
    columnsState: R,
    dataRef: g,
    id: T,
    virtualTotal: ue,
    isAllData: _,
    virtualPageSize: U,
    isVirtualScroll: E,
    detailExpandable: !!e.detail
  };
  return e.scrollable === "none" ? /* @__PURE__ */ a.createElement(et, { ...He }, /* @__PURE__ */ a.createElement(rt, null, /* @__PURE__ */ a.createElement(
    "div",
    {
      id: e.id,
      style: e.style,
      className: f(u.wrapper({ size: I }), e.className),
      ...qe
    },
    he,
    Ce,
    /* @__PURE__ */ a.createElement(
      ul,
      {
        selectable: e.selectable,
        className: f(u.table({ size: I }))
      },
      xe,
      we,
      /* @__PURE__ */ a.createElement(at, { rowReorderable: N }, p),
      Ge
    ),
    Se && /* @__PURE__ */ a.createElement(pe, null, $.length ? $ : /* @__PURE__ */ a.createElement(Ye, null)),
    ke && /* @__PURE__ */ a.createElement(a.Fragment, null, /* @__PURE__ */ a.createElement(tt, null), /* @__PURE__ */ a.createElement(lt, null))
  )), ye, e.pageable && Re, /* @__PURE__ */ a.createElement(it, { loader: e.loader, showLoader: e.showLoader }), De, be && /* @__PURE__ */ a.createElement(Ue, { message: me })) : /* @__PURE__ */ a.createElement(et, { ...He }, /* @__PURE__ */ a.createElement(rt, null, /* @__PURE__ */ a.createElement(
    "div",
    {
      id: e.id,
      style: e.style,
      className: f(
        u.wrapper({
          size: I,
          virtual: E
        }),
        e.className
      ),
      ...qe
    },
    he,
    Ce,
    /* @__PURE__ */ a.createElement(
      "div",
      {
        className: f(u.ariaRoot({})),
        role: "grid",
        "aria-colcount": b.length,
        "aria-rowcount": G,
        id: T,
        "aria-label": O.toLanguageString(nt, yl[nt])
      },
      we,
      /* @__PURE__ */ a.createElement("div", { className: f(u.container({})), role: "presentation" }, /* @__PURE__ */ a.createElement(bl, null, /* @__PURE__ */ a.createElement("div", { className: f(u.content({})), role: "presentation" }, /* @__PURE__ */ a.createElement("div", { className: f(u.tableWrap({})), role: "presentation" }, /* @__PURE__ */ a.createElement(
        gl,
        {
          selectable: e.selectable,
          tableClassName: f(
            u.table({
              size: I
            })
          ),
          tableStyle: { userSelect: Gt }
        },
        xe,
        /* @__PURE__ */ a.createElement(at, { rowReorderable: N }, p)
      ), Se && /* @__PURE__ */ a.createElement(pe, null, $.length ? $ : /* @__PURE__ */ a.createElement(Ye, null))), E && /* @__PURE__ */ a.createElement(
        "div",
        {
          className: f(u.heightContainer({})),
          role: "presentation"
        },
        /* @__PURE__ */ a.createElement(hl, { isVirtualScroll: E })
      )))),
      Ge,
      ke && /* @__PURE__ */ a.createElement(a.Fragment, null, /* @__PURE__ */ a.createElement(tt, null), /* @__PURE__ */ a.createElement(lt, null)),
      be && /* @__PURE__ */ a.createElement(Ue, { message: me })
    ),
    ye,
    e.pageable && Re,
    /* @__PURE__ */ a.createElement(it, { loader: e.loader, showLoader: e.showLoader }),
    De
  )));
});
Al.displayName = "KendoReactGridComponent";
export {
  Al as GridComponent
};
