// @progress/kendo-react-charts/components/base/CollectionConfigurationComponent.mjs
/**
 * @license
 *-------------------------------------------------------------------------------------------
 * Copyright © 2025 Progress Software Corporation. All rights reserved.
 * Licensed under commercial license. See LICENSE.md in the package root for more information
 *-------------------------------------------------------------------------------------------
 */
import * as r from "react";
import { ChartContext as h } from "../../ChartContext.mjs";
const p = class p extends r.Component {
  get optionsStore() {
    var t;
    return (t = this.context) == null ? void 0 : t.optionsStore;
  }
  renderChildren(t, s) {
    const { children: o } = t.props, { _chartKey: c, _parentStore: e } = this.props, n = {
      ...t.props,
      _chartCollectionIdxKey: `${c}_${s}`,
      _parentStore: e
    };
    return r.cloneElement(t, n, o);
  }
  render() {
    const { _chartKey: t, _parentStore: s, children: o } = this.props;
    return (s || this.optionsStore).dispatch({
      chartKey: t,
      payload: []
    }), r.Children.map(o, (e, n) => r.isValidElement(e) ? this.renderChildren(e, n) : e);
  }
};
p.contextType = h;
let a = p;
export {
  a as CollectionConfigurationComponent
};
